docs/code_contribution.md

```diff
@@ -0,0 +1,93 @@
+# Code Contribution Guidelines
+
+To make the EAGER2 code and processing logic more understandable for new contributers, and to ensure quality. We are making an attempt to somewhat-standardise the way the code is written.
+
+If you wish to contribute a new module, please use the following coding standards.
+
+## Process Concept
+
+We are providing a highly configurable pipeline, with many options to turn on and off different processes in different combinations. This can make a very complex graph structure that can cause a large amount of duplicated channels coming out of every process to account for each possible combination.
+
+The EAGER pipeline can currently be broken down into the following 'stages', where a stage is a collection of  non-terminal mututally exclusive processes, which is the output of which is used for another file reporting module (but not reporting!) .
+
+* Input 
```

Comment:
Not sure I like this, makes the pipeline less easy to read 😓  DSLv2 will make this much easier though ...

Reply:
Just for my understanding: What don't you like about it? I agree it's hacky but I can't think of a non-DSLv2 way around it...

We can also remove the page entirely. I thought it would be useful because it was confusing to follow before my changes (and after).

Reply:
It kind of inflates the entire pipeline code - I agree that it would be nice to have all components/stages skippable but the other question would be whether we _really_ need this. Having some points where people can "join" in e.g. with BAM input would be great, but I'd rather have some basic pipeline skipping that just makes certain optional things skippable in general. But again, I'm open to discussing this 👍 

I just find the `if ... else ` stuff around each process quite hacky, thats my only concern here - but I don't see much options to make these things much easier/better (`when` is hard to master without DSLv2 I think...) 

Having something like this page is a good idea I think, to make it easier for new contributors to contribute in general!

Reply:
Fair enough. I put it everywhere because I basically thought though cases I encountered in the past, to see if a skip point at each place would be applicable, which basically was.

Also doesn't EAGER1 effectively also allow that? (BAM input & turning off all modules?). Or does it automatically convert to FASTQ? In that case I could remove the ability to skip mapping. But when I spoke to pop-gen people, they said in many cases nowadays they get 'mapped' BAMs already (to human stuff) - but not with filtering/damageprofiler/qualimap etc, so it would be useful for them.

However, if you want to simplify it - let me know your concept/vision and I can remove skip points where needed.

Reply:
> But when I spoke to pop-gen people, they said in many cases nowadays they get 'mapped' BAMs already (to human stuff) - but not with filtering/damageprofiler/qualimap etc, so it would be useful for them. 

I agree on that, although in the majority of the cases this is bonkers too, because they typically work with already mapped reads there - the unmapped ones are typically dropped, meaning if you remap that you might see some awkward side-effects in mapping, since the unmapped portion might, for example, contain reads mapping against your choice of reference genome. But that is a different story not for this PR 😆 

> Also doesn't EAGER1 effectively also allow that? (BAM input & turning off all modules?). Or does it automatically convert to FASTQ? In that case I could remove the ability to skip mapping. But when I spoke to pop-gen people, they said in many cases nowadays they get 'mapped' BAMs already (to human stuff) - but not with filtering/damageprofiler/qualimap etc, so it would be useful for them.

EAGERv1 does some of this - but doesn't turn off a lot of functionality. I'm not too happy about the BAM input there either, as its rather hacky there too.